packages/medical-domain/src/domains/models/medical/datum/bg.model.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Unit from './enums/unit.enum'
+import type BaseDatum from './basics/base-datum.model'
+import { isBaseDatum } from './basics/base-datum.model'
+import { DatumType } from './enums/datum-type.enum'
+import type WeekDays from '../../time/enum/weekdays.enum'
+
+const MGDL_UNITS = Unit.MilligramPerDeciliter
+const MMOLL_UNITS = Unit.MmolPerLiter
+
+const bgUnits = [MGDL_UNITS, MMOLL_UNITS] as const
+type BgUnit = typeof bgUnits[number]
+
+function isBgUnit(value: unknown): value is BgUnit {
+  if (typeof value === 'string') {
+    return bgUnits.includes(value as BgUnit)
+  }
+  return false
+}
+
+type BgType = DatumType.Cbg | DatumType.Smbg
+
+type Bg = BaseDatum & {
+  type: BgType
+  units: BgUnit
+  value: number
+  // Used for trends view
+  localDate: string
+  isoWeekday: WeekDays
+  msPer24: number
+}
+
+function isBg(value: unknown): value is Bg {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏‏
This kind of function would usually be put under a file such as `bg.util.ts` (and be unit tested)
Same comment for function at line 40

Reply:
Has this comment been forgotten?

Reply:
On my opinion typeguard functions is bound to type and it make sense to keep it in the .model file...

Reply:
Alright :+1: